docs/src/man/comparisons.md

```diff
@@ -1,77 +1,204 @@
+| Complex Function          | `df[['z']].agg(lambda v: np.mean(np.cos(v)))`                                | `combine(df, :z => v -> mean(cos, skipmissing(v)))`           |
+| Transform several columns | `df.agg({'x': max, 'y': min})`                                               | `combine(df, :x => maximum, :y => minimum)`                   |
+|                           | `df[['x','y']].mean()`                                                       | `combine(df, [:x, :y] .=> mean)`                              |
+|                           | `df.filter(regex=("^x")).mean()`                                             | `combine(df, r"^x" .=> mean)`                                 |
+|                           | `df[['x', 'y']].agg([max, min])`                                             | `combine(df, ([:x, :y] .=> [maximum minimum])...)`            |
+| Multivariate function     | `df.assign(x_y_cor = np.corrcoef(df.x, df.y)[0,1])`                          | `transform(df, [:x, :y] => cor)`                              |
+| Row-wise                  | `df.assign(x_y_min = df.apply(lambda v: min(v.x, v.y), axis=1))`             | `transform(df, [:x, :y] => ByRow(min))`                       |
+|                           | `df.assign(x_y_argmax = df.apply(lambda v: df.columns[v.argmax()], axis=1))` | `transform(df, AsTable([:x,:y]) => ByRow(argmax))`            |
+| DataFrame as input        | `df.groupby('grp').head(2)`                                                  | `combine(d -> first(d, 2), groupby(df, :grp))`                |
+| DataFrame as output       | `df[['x']].agg(['min','max'])`                                               | `combine(:x => x -> (value = [minimum(x), maximum(x)],), df)` |
+
+### Joining data frames
+
+Suppose that you have a second data frame as shown below:
+```
+df2 = pandas.DataFrame({'grp': [1], 'w': [10]})
+
+>>> df2
+   grp   w
+0    1  10
+```
+
+Here is how to join the data frames:
+
+| Operation             | pandas                                         | DataFrames.jl                   |
+| :-------------------- | :--------------------------------------------- | :------------------------------ |
+| Inner join            | `pd.merge(df, df2, how = 'inner', on = 'grp')` | `innerjoin(df, df2, on = :grp)` |
+| Outer join            | `pd.merge(df, df2, how = 'outer', on = 'grp')` | `outerjoin(df, df2, on = :grp)` |
+| Left join             | `pd.merge(df, df2, how = 'left', on = 'grp')`  | `leftjoin(df, df2, on = :grp)`  |
+| Right join            | `pd.merge(df, df2, how = 'right', on = 'grp')` | `rightjoin(df, df2, on = :grp)` |
+| Semi join (filtering) | `df[df.grp.isin(df2.grp)]`                     | `semijoin(df, df2, on = :grp)`  |
+| Anti join (filtering) | `df[~df.grp.isin(df2.grp)]`                    | `antijoin(df, df2, on = :grp)`  |
```

Comment:
I think these two examples are misleading, as they seem to only work for one column when you do a join in pandas. How to make it work for multiple columns you would want to group by?

Reply:
This is where DataFrames.jl shines as it's quite ugly in pandas 🤯 

```
>>> df
   grp  x  y    z
a    1  6  4  3.0
b    2  5  5  4.0
c    1  4  6  5.0
d    2  3  7  6.0
e    1  2  8  7.0
f    2  1  9  NaN
>>> df2
   grp  x   w
0    1  6  10
>>> df[df[['grp','x']].agg(tuple,1).isin(df2[['grp','x']].agg(tuple,1))]
   grp  x  y    z
a    1  6  4  3.0
```

I reckon that this is not a very common join operation after searching on Stack Overflow. What do you think of this commit https://github.com/JuliaData/DataFrames.jl/pull/2378/commits/a73e8981c3884f1ba8b536d62ded47f7c6b530d6 ?

Reply:
OK - we can leave it as is.